src/auth.ts

```diff
@@ -0,0 +1,113 @@
+import express from 'express';
+import crypto from 'crypto';
+import cookieParser from 'cookie-parser';
+// import cors from 'cors';
+import { URLSearchParams } from 'url';
+import { connectDB, storeRefreshToken } from './db';
+import { ngrokURL, spotify } from './config.json';
+import logger from './utils/logger';
+
+const client_id = spotify.clientId;
+const client_secret = spotify.clientSecret;
+const redirect_uri = ngrokURL + '/callback';
+
+const stateKey = 'spotify_auth_state';
+
+connectDB();
+
+const generateRandomString = (length: number): string => {
+    return crypto
+        .randomBytes(60)
+        .toString('hex')
+        .slice(0, length);
+};
+
+const app = express();
+app.use(cookieParser());
+
+app.get('/login', (req, res) => {
+    const state = generateRandomString(16);
+    const { userId } = req.query;
+    res.cookie(stateKey, state);
+    res.cookie("userId", userId);
+
+    const scope = 'user-read-private user-read-email';
```

Comment:
Okay my bad for not mentioning this before, but we're gonna need more scopes, specifically these:
`'user-read-currently-playing user-read-playback-state user-modify-playback-state user-read-playback-position playlist-read-collaborative playlist-read-private'`

Reply:
OK